app/build.gradle

```diff
@@ -1,306 +0,0 @@
-                it.contains("jacocoTestReport")
-            }
-            applicationIdSuffix = ".test"
-            resValue "string", "search_provider_authority", android.defaultConfig.applicationId + applicationIdSuffix + ".SuggestionsProvider"
-        }
-        release {
-            minifyEnabled true
-            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
-            resValue "string", "search_provider_authority", android.defaultConfig.applicationId + ".SuggestionsProvider"
-        }
-    }
-
-    testOptions {
-        // Uncomment below to use orchestrator for tests
-        // execution 'ANDROIDX_TEST_ORCHESTRATOR'
-        unitTests {
-            all {
-                jvmArgs '-noverify', '-ea'
-                jacoco {
-                    includeNoLocationClasses = true
-                    jacoco.excludes = ['jdk.internal.*']
-                }
-            }
-            includeAndroidResources = true
-        }
-    }
-
-    if (rootProject.hasProperty("AndroidSigningKeyAlias")
-            && rootProject.hasProperty("AndroidSigningKeyPassword")
-            && rootProject.hasProperty("AndroidSigningStoreFile")
-            && rootProject.hasProperty("AndroidSigningStorePassword")) {
-        println "Using signing properties from gradle properties"
-        signingConfigs {
-            release {
-                keyAlias rootProject.AndroidSigningKeyAlias
-                keyPassword rootProject.AndroidSigningKeyPassword
-                storeFile file(rootProject.AndroidSigningStoreFile)
-                storePassword rootProject.AndroidSigningStorePassword
-            }
-            android.buildTypes.release.signingConfig = signingConfigs.release
-        }
-    } else if (System.env.AndroidSigningKeyPassword != null
-                && System.env.AndroidSigningKeyPassword != null
-                && System.env.AndroidSigningStoreFile != null
-                && System.env.AndroidSigningStorePassword != null) {
-        println "Using signing properties from environment variables"
-            signingConfigs {
-                release {
-                    keyAlias System.env.AndroidSigningKeyAlias
-                    keyPassword System.env.AndroidSigningKeyPassword
-                    storeFile file(System.env.AndroidSigningStoreFile)
-                    storePassword System.env.AndroidSigningStorePassword
-                }
-                android.buildTypes.release.signingConfig = signingConfigs.release
-            }
-    } else {
-        println "No signing properties found"
-        android.buildTypes.release.signingConfig = null
-    }
-}
-
-jacoco {
-    toolVersion '0.8.12'
-}
-android.applicationVariants.all { variant ->
-    def copyLicenseFilesTask = tasks.register("copyLicenseFilesFor${variant.name.capitalize()}", Copy) {
-        from project.rootDir
-        into "${project.buildDir}/generated/license_assets/"
```

Comment:
`project.buildDir` as a string has been replaced with `project.layout.buildDirectory` which is a directory object.